src/vos/vos_pool.c

```diff
@@ -616,6 +616,16 @@ vos_register_slabs(struct umem_attr *uma)
 			return rc;
 		}
 
+		for (j = 0; j < i; j++) {
+			if (uma->uma_slabs[j].unit_size == slab->unit_size) {
+				/** PMDK will fail to register a new slab of the same size
+				 *  so reuse the class id
+				 */
+				slab->class_id = uma->uma_slabs[j].class_id;
+				goto skip_new_class;
+			}
+		}
```

Comment:
I know we still have many of goto in loop, but we should avoid them unless it can significantly simply code, e.g., we can do this:
```
bool skip_new_class = false;

for (j = 0; j < i; j++) {
    bool skip_new_class = false;
    if (uma->....) {
        slab->class_id = ...;
        skip_new_class = true;
        break;
    }
    if (skip_new_class) {
          D_ASSERT(slab->class_id != 0);
          continue;
    }
```

Reply:
Fair enough...if I refresh the pass, I will fix this.

I never used goto until I joined this team